cmake/list_insert_options.cmake

```diff
@@ -0,0 +1,55 @@
+#
+# Append key[=value] elements from one list to another list if they are not already in the target list.
```

Comment:
If the same key is being inserted multiple time with different values shouldn't the last one be used (instead of the first one)?